src/utils/management/CustomState.ts

```diff
@@ -1,34 +1,23 @@
-export class CustomState<T> {
-  private error: string;
-  private pending: boolean;
-  private data: T;
-  private readonly preparedData: T;
+export function State<T>(data: T): CustomState<T> {
+  return new CustomState(data);
+}
 
-  constructor(preparedData: T) {
-    this.preparedData = preparedData;
-  }
+class CustomState<T> {
+  error: string;
+  pending: boolean;
+  preparedData: T;
+
+  constructor(public data: T) {}
```

Comment:
You need to save initial data for init purposes it's obvious. But still, we can improve it a little bit.
Change public data: T - to private _data: T - this hides initial state object passed to constructor.

Rename prepadedData: T - to data: T.

Inside init - replace this.prepadedData - with this._data.